spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/TraceWebFilter.java

```diff
@@ -235,18 +264,40 @@ private Span findOrCreateSpan(Context c) {
 
 			final HttpServerHandler<HttpServerRequest, HttpServerResponse> handler;
 
+			final CurrentTraceContext currentTraceContext;
+
+			Subscription s;
+
 			WebFilterTraceSubscriber(CoreSubscriber<? super Void> actual, Context context,
 					Span span, MonoWebFilterTrace parent) {
 				this.actual = actual;
 				this.span = span;
 				this.context = context.put(TraceContext.class, span.context());
 				this.exchange = parent.exchange;
 				this.handler = parent.handler;
+				this.currentTraceContext = parent.currentTraceContext;
 			}
 
 			@Override
 			public void onSubscribe(Subscription subscription) {
-				this.actual.onSubscribe(subscription);
+				this.s = subscription;
+				try (Scope scope = this.currentTraceContext.maybeScope(span.context())) {
+					this.actual.onSubscribe(this);
+				}
+			}
+
+			@Override
+			public void request(long n) {
+				try (Scope scope = this.currentTraceContext.maybeScope(span.context())) {
+					this.s.request(n);
+				}
+			}
+
+			@Override
+			public void cancel() {
+				try (Scope scope = this.currentTraceContext.maybeScope(span.context())) {
```

Comment:
do we expect overhead like this to be worth it? operations that quickly return? I'm not saying I think it shouldn't but each of  "trigger type commands" you can ask the question as if for example, the act of doing that does only small local code it can be unhelpful to wrap.